as3935_i2c binary_sensor.h: No such file or directory

### The problem

When compiling using the [as3935 sensor](https://esphome.io/components/sensor/as3935.html), it fails with the below error message:
```
In file included from src\esphome\components\as3935_i2c\as3935_i2c.cpp:1:
src\esphome\components\as3935_i2c\as3935_i2c.h:7:10: fatal error: esphome/components/binary_sensor/binary_sensor.h: No such file or directory
    7 | #include "esphome/components/binary_sensor/binary_sensor.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
*** [.pioenvs\sen54-test\src\esphome\components\as3935_i2c\as3935_i2c.cpp.o] Error 1
```

### Which version of ESPHome has the issue?

2025.4.1

### What type of installation are you using?

pip

### Which version of Home Assistant has the issue?

_No response_

### What platform are you using?

ESP8266

### Board

d1_mini_lite

### Component causing the issue

as3935

### YAML Config

```yaml
esphome:
  name: sen54-test

esp8266:
  board: d1_mini_lite
  framework:
    version: recommended

logger:

api:
  password: ""

ota:
  - platform: esphome
    password: ""

web_server:
  port: 80

sensor:
  - platform: as3935
    lightning_energy:
      name: Lightning Energy
    distance:
      name: Distance Storm

as3935_i2c:
  irq_pin: GPIO16

i2c:
  sda: GPIO4
  scl: GPIO5
  scan: true
  id: bus_a
  frequency: 200kHz

wifi:
  ssid: "REDACTED"
  password: "REDACTED"

captive_portal:
```

### Anything in the logs that might be useful for us?

```txt
In file included from src\esphome\components\as3935_i2c\as3935_i2c.cpp:1:
src\esphome\components\as3935_i2c\as3935_i2c.h:7:10: fatal error: esphome/components/binary_sensor/binary_sensor.h: No such file or directory
    7 | #include "esphome/components/binary_sensor/binary_sensor.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
*** [.pioenvs\sen54-test\src\esphome\components\as3935_i2c\as3935_i2c.cpp.o] Error 1
=============================================================================================== [FAILED] Took 11.45 seconds ===============================================================================================
```

### Additional information

  **System:**
    OS: Windows 11 10.0.26100
    CPU: (4) x64 Intel(R) N100
    Memory: 4.98 GB / 11.78 GB
  **Binaries:**
    Node: 22.15.0 - C:\nvm4w\nodejs\node.EXE
    npm: 10.9.2 - C:\nvm4w\nodejs\npm.CMD
  **Managers:**
    pip3: 25.1.1 - ~\AppData\Local\Programs\Python\Python313\Scripts\pip3.EXE
  **Utilities:**
    Git: 2.49.0.
    Curl: 8.12.1 - C:\WINDOWS\system32\curl.EXE
  **IDEs:**
    VSCode: 1.99.3 - C:\Users\XXXXX\AppData\Local\Programs\Microsoft VS Code\bin\code.CMD
  **Languages:**
    Python: 3.13.3
  **Browsers:**
    Edge: Chromium (133.0.3065.92)

Command to reproduce:
```bash
 esphome compile .\test.yaml      
```

--- a/esphome/components/as3935_i2c/as3935_i2c.h
+++ b/esphome/components/as3935_i2c/as3935_i2c.h
@@ -1,10 +1,7 @@
 #pragma once
 
-#include "esphome/core/component.h"
 #include "esphome/components/as3935/as3935.h"
 #include "esphome/components/i2c/i2c.h"
-#include "esphome/components/sensor/sensor.h"
-#include "esphome/components/binary_sensor/binary_sensor.h"
 
 namespace esphome {
 namespace as3935_i2c {

--- a/tests/components/as3935_i2c/test.esp32-ard.yaml
+++ b/tests/components/as3935_i2c/test.esp32-ard.yaml
@@ -3,4 +3,9 @@ substitutions:
   sda_pin: GPIO17
   irq_pin: GPIO15
 
-<<: !include common.yaml
+packages:
+  as3935: !include common.yaml
+
+# Trigger issue: https://github.com/esphome/issues/issues/6990
+# Compile with no binary sensor results in error
+binary_sensor: !remove

